src/test/java/org/mockitousage/junitrule/StrictJUnitRuleTest.java

```diff
@@ -0,0 +1,182 @@
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                mock.simpleMethod();
+                Mockito.verify(mock).simpleMethod();
+            }
+        });
+    }
+
+    @Test public void ok_when_all_stubbings_used() throws Throwable {
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock1, IMethods mock2) throws Throwable {
+                IMethods mock = mock(IMethods.class);
+                given(mock.simpleMethod(10)).willReturn("foo");
+                mock.simpleMethod(10);
+            }
+        });
+    }
+
+    @Test public void ok_when_used_and_mismatched_argument() throws Throwable {
+        run(new MockitoStatement() {
+            public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                given(mock.simpleMethod(10)).willReturn("foo");
+                mock.simpleMethod(10);
+                mock.simpleMethod(15);
+            }
+        });
+    }
+
+    @Test public void fails_when_unused_stubbings() throws Throwable {
+        try {
+            //when
+            run(new MockitoStatement() {
+                public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                    given(mock.simpleMethod(10)).willReturn("foo");
+                    mock2.simpleMethod(15);
+                }
+            });
+
+            //then
+            fail();
+        } catch (MockitoAssertionError e) {
+            Assertions.assertThat(e.getMessage()).startsWith("Unused stubbings");
+        }
+    }
+
+    @Test public void test_failure_trumps_unused_stubbings() throws Throwable {
+        try {
+            //when
+            run(new MockitoStatement() {
+                public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                    given(mock.simpleMethod(10)).willReturn("foo");
+                    mock.otherMethod();
+
+                    throw new AssertionError("x");
+                }
+            });
+
+            //then
+            fail();
+        } catch (AssertionError e) {
+            assertEquals("x", e.getMessage());
+        }
+    }
+
+    @Test public void fails_fast_when_stubbing_invoked_with_different_argument() throws Throwable {
+        try {
+            //when
+            run(new MockitoStatement() {
+                public void evaluate(IMethods mock, IMethods mock2) throws Throwable {
+                    //stubbing in the test code:
+                    given(mock.simpleMethod(10)).willReturn("foo");
+
+                    //invocation in the code under test uses different argument and should fail immediately
+                    //this helps with debugging and is essential for Mockito strictness
+                    mock.simpleMethod(15);
+                }
+            });
+
+            //then
+            fail();
+        } catch (MockitoAssertionError e) {
+            e.printStackTrace();
```

Comment:
Left-over statement. You can prevent this using `ExpectedException` as pointed out by @ChristianSchwarz